.github/scripts/apidoc.sh

```diff
@@ -0,0 +1,158 @@
+
+if [ -z "$APIDOC_TOKEN" ]; then
+    echo "No APIDOC_TOKEN available. Skipping."
+    exit
+fi
+
+# NOTE: DO NOT USE "set -x", or anything else that would reveal APIDOC_TOKEN!
+# GitHub Actions does filter secrets, but extra caution won't hurt
+set -e
+set +x
+
+# Display exit code in case of failure (probably due to 'set -x').
+trap '[ "$?" = 0 ] || echo "EXIT CODE: $?"' EXIT
+
+REPO_APIDOC="https://${APIDOC_TOKEN}@github.com/awesomeWM/apidoc"
+REPO_DIR="$PWD"
+
+export GIT_AUTHOR_NAME="awesome-robot on GitHub Actions"
+export GIT_AUTHOR_EMAIL="awesome-robot@users.noreply.github.com"
+export GIT_COMMITTER_NAME="$GIT_AUTHOR_NAME"
+export GIT_COMMITTER_EMAIL="$GIT_AUTHOR_EMAIL"
+
+git clone --depth 1 --branch gh-pages "$REPO_APIDOC" build/apidoc \
+    2>&1 | sed "s/$APIDOC_TOKEN/APIDOC_TOKEN/g"
+cd build/apidoc
+
+# This will re-use already existing branches (updated PR).
+if [ "$PR_NUMBER" != false ]; then
+    BRANCH="pr-$PR_NUMBER"
+elif [ "$GITHUB_HEAD_REF" != master ]; then
+    # Use merge-base of master in branch name, to keep different branches with
+    # the same name apart.
+    # shellcheck disable=SC2015
+    BRANCH="$GITHUB_HEAD_REF-$(cd "$REPO_DIR" \
+        && git fetch --unshallow origin master \
+        && git rev-parse --short "$(git merge-base HEAD FETCH_HEAD || true)" || true)"
+else
+  BRANCH="gh-pages"
+fi
+if [ "$BRANCH" != "gh-pages" ]; then
+    git checkout -b "$BRANCH" "origin/${BRANCH}" 2> /dev/null || git checkout -b "$BRANCH"
+fi
+
+# Use a temporary branch for the two commits, which allows for a better UI.
+git checkout -b merged-update
+
+# Create the README for the Git repo (https://github.com/awesomeWM/apidoc).
+cat > ../doc/README.md <<END
+# Awesome API documentation
+
+This repository contains the built API documentation for the
+[awesome](https://github.com/awesomeWM/awesome) window manager. It is
+automatically updated via Travis when the master branch changes. Hence:
+
+## Do NOT send pull requests here
+
+Instead, please update the source code of
+[awesome](https://github.com/awesomeWM/awesome) instead.
+END
+
+# Create a patch without irrelevant changes (version / timestamp).
+diff -Nur . ../doc -I "Last updated" -I "<strong>Release</strong>:" \
+    -I "<h2>API documentation for awesome, a highly configurable X window manager (version .*)\.</h2>" \
+    -x .git | patch -p1
+
+git add --all .
+if git diff --cached --exit-code --quiet; then
+    echo "Documentation has not changed."
+    exit
+fi
+
+LAST_COMMIT_MSG="$(cd "$REPO_DIR" && git log -1 --pretty=format:%s)"
+LAST_COMMIT="$(cd "$REPO_DIR" && git rev-parse --short HEAD)"
+
+# Commit the relevant changes.
+COMMIT_MSG="Update docs for $AWESOME_VERSION via Github Actions
+
+Last commit message:
+$LAST_COMMIT_MSG
+
+Commits: https://github.com/awesomeWM/awesome/compare/${GITHUB_BASE_REF}..${GITHUB_HEAD_REF}
+Build URL: $GITHUB_SERVER_URL/$GITHUB_REPOSITORY/actions/runs/$GITHUB_RUN_ID"
+git commit -m "[relevant] $COMMIT_MSG"
```

Comment:
nit, can you insert an empty line between lines 95 and 96 ?
This would help to identify lines 89-95 as a single quoted block